config.toml

```diff
@@ -14,3 +14,7 @@ OPENAI = "<YOUR_OPENAI_API_KEY>"
 
 [API_ENDPOINTS]
 BING = "https://api.bing.microsoft.com/v7.0/search"
+OPENAI_BASE_URL = "https://api.openai.com/v1"
+
+[USER_PREFS]
+SEARCH_ENGINE = "Bing" # Possible Values: Bing, DDGS. Default Value: Bing
```

Comment:
**Security Issue**: Adding a new configuration for the OpenAI API base URL and user preferences for search engines in plain text to the configuration file can expose the application to configuration tampering and potentially redirect sensitive operations to unauthorized endpoints. <br> **Fix**: Store sensitive configuration details such as API base URLs and user preferences in an encrypted format. Additionally, implement checksum verification for the configuration file to detect unauthorized modifications. <br> **Code Suggestion**: 
 ```
 +# It is recommended to store sensitive information in an encrypted format and implement checksum verification
 +# for the configuration file to prevent unauthorized modifications.
 +# This example demonstrates a placeholder approach and should be replaced with actual encryption and checksum mechanisms.
 +OPENAI_BASE_URL = encrypt("https://api.openai.com/v1")
 +SEARCH_ENGINE = encrypt("Bing")
 +config_checksum = generate_checksum(config_file_contents)
 ```